### PR TITLE
[telegram] Functionality added to delete a Query

### PR DESCRIPTION
**Pull request by fd0cwp**
I have added the possibility to remove a telegram query from a chat.

For details, please see #15716.

Signed-off-by: Christoph <fd0cwp@gmx.de>

### CODE REVIEW

File: bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/action/TelegramActions.java [hunk 1]
```diff
@@ -636,6 +670,14 @@ public static boolean sendTelegramAnswer(ThingActions actions, @Nullable String
         return ((TelegramActions) actions).sendTelegramAnswer(replyId, message);
     }
 
+    public static boolean deleteTelegramQuery(ThingActions actions, @Nullable String replyId) {
+        if (actions instanceof TelegramActions) {
+            return ((TelegramActions) actions).deleteTelegramQuery(replyId);
```
Comment (lsiepel) [inline bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/action/TelegramActions.java:675]:
```suggestion
        if (actions instanceof TelegramActions telegramActions) {
            return telegramActions.deleteTelegramQuery(replyId);
```

File: bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/action/TelegramActions.java [hunk 2]
```diff
@@ -283,6 +284,39 @@ private boolean sendTelegramGeneral(@ActionInput(name = "chatId") @Nullable Long
...
+        for (Long chatId : localHandler.getReceiverChatIds()) {
+            messageId = localHandler.removeMessageId(chatId, replyId);
+            if (messageId == null) {
+                logger.debug("deleteTelegramQuery() - messageId not found!");
+                return false;
+            }
+            DeleteMessage deleteMessage = new DeleteMessage(chatId, messageId);
+            response = localHandler.execute(deleteMessage);
+
+            if ((response == null) || (response.errorCode() != 0)) {
```
Comment (lsiepel) [inline bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/action/TelegramActions.java:311]:
```suggestion
            if (response == null || response.errorCode() != 0) {
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Some minor comments

**Comment by fd0cwp** [comment]
Thank you for your quick review, @lsiepel!

Why is it reasonable to remove the brackets? In my opinion, it is clearer and easier to read with the brackets.

File: bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/action/TelegramActions.java (hunk 1 above)
Comment (fd0cwp) [reply bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/action/TelegramActions.java]: I made the same changes in my local 4.1.1 branch, and there it shows an error: telegramActions cannot be resolved
This is valid in my point of view, but why does it work in the main branch? 🤔

**Comment by fd0cwp** [comment]
And how do I get rid of the listed Commits on Apr 6, 2024 regarding the intesis changes?

**Review by lsiepel — APPROVED** [review]
LGTM

**Pull request merged**